lib/components/OrderDialog/OrderDialog.tsx

```diff
@@ -1,58 +1,70 @@
+import { CheckoutOrder } from "./CheckoutOrder"
+import { InitialOrderScreen } from "./InitialOrder"
+import { type Step, StepwiseProgressPanel } from "./StepwiseProgress"
 
 interface OrderDialogProps {
   isOpen: boolean
   onClose: () => void
+  stage: "initial" | "progress" | "checkout"
+  setStage: (stage: "initial" | "progress" | "checkout") => void
 }
 
-const orderSteps = [
-  { key: "are_gerbers_generated", label: "Generate Gerbers" },
-  { key: "are_gerbers_uploaded", label: "Upload Gerbers" },
-  { key: "is_gerber_analyzed", label: "Analyze Gerber" },
-  { key: "are_initial_costs_calculated", label: "Calculate Initial Costs" },
-  { key: "is_pcb_added_to_cart", label: "Add PCB to Cart" },
-  { key: "is_bom_uploaded", label: "Upload BOM" },
-  { key: "is_pnp_uploaded", label: "Upload PnP" },
-  { key: "is_bom_pnp_analyzed", label: "Analyze BOM & PnP" },
-  { key: "is_bom_parsing_complete", label: "BOM Parsing Complete" },
-  { key: "are_components_available", label: "Components Available" },
-  { key: "is_patch_map_generated", label: "Generate Patch Map" },
-  { key: "is_json_merge_file_created", label: "Create JSON Merge File" },
-  { key: "is_dfm_result_generated", label: "Generate DFM Result" },
-  { key: "are_files_downloaded", label: "Download Files" },
+const orderSteps: Step[] = [
+  { id: 1, key: "are_gerbers_generated", title: "Generate Gerbers" },
+  { id: 2, key: "are_gerbers_uploaded", title: "Upload Gerbers" },
+  { id: 3, key: "is_gerber_analyzed", title: "Analyze Gerber" },
   {
+    id: 4,
+    key: "are_initial_costs_calculated",
+    title: "Calculate Initial Costs",
+  },
+  { id: 5, key: "is_pcb_added_to_cart", title: "Add PCB to Cart" },
+  { id: 6, key: "is_bom_uploaded", title: "Upload BOM" },
+  { id: 7, key: "is_pnp_uploaded", title: "Upload PnP" },
+  { id: 8, key: "is_bom_pnp_analyzed", title: "Analyze BOM & PnP" },
+  { id: 9, key: "is_bom_parsing_complete", title: "BOM Parsing Complete" },
+  { id: 10, key: "are_components_available", title: "Components Available" },
+  { id: 11, key: "is_patch_map_generated", title: "Generate Patch Map" },
+  {
+    id: 12,
+    key: "is_json_merge_file_created",
+    title: "Create JSON Merge File",
+  },
+  { id: 13, key: "is_dfm_result_generated", title: "Generate DFM Result" },
+  { id: 14, key: "are_files_downloaded", title: "Download Files" },
+  {
+    id: 15,
     key: "are_product_categories_fetched",
-    label: "Fetch Product Categories",
+    title: "Fetch Product Categories",
+  },
+  { id: 16, key: "are_final_costs_calculated", title: "Calculate Final Costs" },
+  {
+    id: 17,
+    key: "is_json_merge_file_updated",
+    title: "Update JSON Merge File",
   },
-  { key: "are_final_costs_calculated", label: "Calculate Final Costs" },
-  { key: "is_json_merge_file_updated", label: "Update JSON Merge File" },
-  { key: "is_added_to_cart", label: "Add to Cart" },
-]
+  { id: 18, key: "is_added_to_cart", title: "Add to Cart" },
```

Comment:
we never use `id` (always prefix an id, it prevents many, many bugs)

in this case `order_step_id` or `orderStepId`